### DISCUSSION TITLE
我不信你大學沒用過 Stack Overflow

### DISCUSSION THREAD

**Discussion by lyykey** [1101-GS4719-C/Python-Community-GS4719-C#1913] [General]
Stack Overflow 是一個程式設計領域的問答網站。打從我上大學到現在，每當遇到程式語法的問題，或是環境的編譯問題。除了翻書，更多的是上網找資料，而通常顯示的前幾項網頁就會有 Stack Overflow。
當然除了問題解決，你也可以在這個網站上學習到很多關於計算機知識的內容，像是資料結構、演算法的知識。
以下為網站連結：[https://stackoverflow.com/](https://stackoverflow.com/)
以前曾遇過的案件如下:
執行一項專案，有套件沒有安裝到作業系統，把問題上傳到 google，第一個跳出的網頁會是 Stack Overflow，而且通常能替我解決問題。
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/105580604/adf1a19f-4637-4658-bee0-b2e393ad1c61)

我認為 Stack Overflow 是一個非常有價值的網站，相信身為一個大學生都會利用這個網站解決你的程式問題，希望這個資訊對大家有幫助。

**Comment by OG-Matcha** [comment]
我覺得 Stack Overflow 真的是聖經，幾乎遇過的問題在上面都能找到，就算現在有 ChatGPT 等工具，我還是很常直接去 Google 找Stack Overflow 上的解法。

**Comment by benbebobo149** [comment]
我認為不是每個人都會利用Stack Overflow，因為有些人可能根本不知道這樣的資源存在。另外，隨著生成式AI如ChatGPT的崛起，它們使用Stack Overflow和Reddit等網站作為訓練數據，這導致許多人直接使用ChatGPT等工具，而不再頻繁使用Stack Overflow，這也給這些網站帶來了巨大的流量壓力。我覺得這種科技的興起很可能會改變人們獲取資訊的方式，這讓我對Stack Overflow的未來有些擔憂，但同時也期待它能夠做出適應性的改變，以因應這個變化的時代。感謝同學的分享，讓我們一起思考這些議題。

**Comment by William08290829** [comment]
在使用Stack Overflow前，記得先了解該如何提出清晰明確的問題
我自己以前不了解就直接上去問過問題，然後後來被一個網友"強烈叮嚀"直接被投downvote  :(
問問題真的是一種技巧，要提供足夠的上下文，例如程式碼的片段、操作環境版本資訊，以及嘗試解決問題的具體步驟，讓其他可以迅速的理解你的問題。
還有記得要禮貌表達問題，但是不是說直接把程式碼丟上去, 其他內容也沒打詳細
然後最後留一個謝謝, 這樣不叫做禮貌喔ww

**Comment by KH1126** [comment]
stack overflow是個很好的網站，當編譯出問題或是有其他error時，
通過在stack overflow將error的資訊進行搜索通常都可以在上面找到解決辦法。
因此stack overflow可說是開發者的福音。

**Comment by yyujimmy** [comment]
我相信Stack Overflow 絕對是許多開發者必備的利器之一！我也常常在需要解決問題或尋找特定資訊時造訪這個網站。我喜歡它的搜索功能，能夠快速找到我需要的答案。而且，Stack Overflow 的社群非常活躍，這意味著我可以在那裡找到豐富且多樣的見解和解決方案。另外，我也喜歡它的回答評級系統，這有助於我找到高品質的解答。總之，Stack Overflow 在我解決問題和學習新知識方面一直是一個非常寶貴的資源。

**Comment by wanderer657** [comment]
我發現Stack Overflow就像是程式設計的黃金寶藏，每當我被程式碼搞得頭痛時，它就像是個神一樣的存在。相對於其他資源，它提供的解答真的就是一絕，讓我解決了無數次的編譯問題和套件安裝困擾。說真的，沒有Stack Overflow，我可能早就被程式碼玩到崩潰了。所以，我覺得每個大學生都應該拜它為師，毫不客氣地說，Stack Overflow就是我的救星！

**Comment by LinChloe** [comment]
我自從跑出error時，常常看它參考許多先人的解決辦法。因為上面許多人已經幫我把該踩的雷都踩完了，導致我遇到問題都不用提問，已經有人提問並解決完畢了。老實說比起chatgpt，我更怕它消失不見，chatgpt有時會胡言亂語，氣到我想把電腦摔了。但這網站上面的人都非常聰明，不會亂回答我的問題。
適用時機:
error、原文書習題

**Comment by wanka2030** [comment]
Stack Overflow真的是在寫程式的救星，在最一開始要安裝C++的編譯氣在VSCode中總會搞的人仰馬翻，總是會因為找不到所保存的位置在哪裡，又或是找不到Path的位置要放在哪裡。但那時候除了利用Stack Overflow之外，也有去問問看Chatgpt有什麼辦法，但大多數時間前者能夠給的feedback有比較多，也較為專業。

**Comment by YANRONG0123** [comment]
正如作者所言，Stack Overflow 不僅僅是一個問答網站，更是一個充滿知識和學習機會的社群。若我們充分利用這個寶貴的資源，不僅能夠解決眼前的問題，還能夠不斷地提升自己的技術水平。感謝作者分享這篇寶貴的文章，我相信對許多程式設計者來說都會非常有幫助。感謝您讓我這個其他理工科系的學生知道有這樣的資源。
